libnmstate/netapplier.py

```diff
@@ -33,12 +36,31 @@ def _apply_ifaces_state(state):
 
     for iface_state in state['interfaces']:
         nmdev = client.get_device_by_iface(iface_state['name'])
+        active_connection = nmdev.get_active_connection()
+
         if iface_state['state'] == 'up':
-            if nmdev.get_state() == nmclient.NM.DeviceState.ACTIVATED:
-                continue
-            client.activate_connection_async(device=nmdev)
+            ip4state = iface_state.get("ip")
```

Comment:
please keep the IP logic in separate functions, this would get really messy once we add support for more features